libs/state/src/lib/cell-summary/cell-summary.state.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable } from '@angular/core';
+import { Action, State, StateContext } from '@ngxs/store';
+import { z } from 'zod';
+import { ComputeAggregate } from './cell-summary.actions';
+import { AggregateRow, CellSummaryStateModel, CELL_SUMMARY_AGGREGATE_SCHEMA } from './cell-summary.model';
+
+/** State handling cell summary data */
+@State<CellSummaryStateModel>({
+  name: 'cellsummary',
+  defaults: {
+    summaries: {},
+    aggregate: {},
+  },
+})
+@Injectable()
+export class CellSummaryState {
+  /**
+   * computes aggregate data and stores in the current state
+   */
+  @Action(ComputeAggregate)
+  computeAggregate(ctx: StateContext<CellSummaryStateModel>, action: ComputeAggregate) {
+    const summaries = action.summaries;
+    const aggregateData: z.infer<typeof CELL_SUMMARY_AGGREGATE_SCHEMA> = {};
+
+    for (const key in summaries) {
```

Comment:
The logic inside this loop is hard to follow because of the deep nesting. Break it apart into private methods on the class will help with readability